dm/pkg/ha/openapi_task_config.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ha
+
+import (
+	"context"
+
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/clientv3/clientv3util"
+
+	"github.com/pingcap/ticdc/dm/dm/common"
+	"github.com/pingcap/ticdc/dm/openapi"
+	"github.com/pingcap/ticdc/dm/pkg/etcdutil"
+	"github.com/pingcap/ticdc/dm/pkg/terror"
+)
+
+func openAPITaskFromResp(resp *clientv3.GetResponse) (*openapi.Task, error) {
+	task := &openapi.Task{}
+	if resp.Count == 0 {
+		return nil, nil
+	} else if resp.Count > 1 {
+		// this should not happen.
+		return task, terror.ErrConfigMoreThanOne.Generate(resp.Count, "openapi.Task", "")
+	}
+	// we make sure only have one task config.
+	if err := task.FromJSON(resp.Kvs[0].Value); err != nil {
+		return task, err
+	}
+	return task, nil
+}
+
+// PutOpenAPITaskConfig puts the openapi task config of task-name.
+func PutOpenAPITaskConfig(cli *clientv3.Client, task openapi.Task, overWrite bool) error {
```

Comment:
I prefer to wrap this to two new func with `overWrite=true` and `overWrite=false`.

Just a suggestion.

Reply:
i think use paramaters is enough